lms/djangoapps/program_enrollments/api/reading.py

```diff
@@ -377,6 +382,41 @@ def get_users_by_external_keys(program_uuid, external_user_keys):
     return users_by_external_keys
 
 
+def get_external_key_by_user_and_course(user, course_key):
+    """
+    Returns the external_user_key of the edX account/user
+    enrolled into the course
+
+    Arguments:
+        user (User):
+            The edX account representing the user in auth_user table
+        course_key (CourseKey|str):
+            The course key of the course user is enrolled in
+
+    Returns: external_user_key (str|None)
+        The external user key provided by Masters degree provider
+        Or None if cannot find edX user to Masters learner mapping
+    """
+    program_course_enrollments = ProgramCourseEnrollment.objects.filter(
+        course_enrollment__user=user,
+        course_key=course_key
+    ).order_by('-modified')
+
+    if not program_course_enrollments:
+        return None
+
+    program_enrollment = None
+    for program_course_enrollment in program_course_enrollments:
+        program_enrollment = program_course_enrollment.program_enrollment
+        if program_course_enrollment.status == ProgramCourseEnrollmentStatuses.ACTIVE:
```

Comment:
Why can't we filter for this at the query level?

Reply:
I imagine that Simon wants us to fall back to an inactive program course enrollment if there are no active ones. Do I have that right @schenedx ?

Reply:
@kdmccormick's explanation is correct